src/table/hooks/use-column-widths.ts

```diff
@@ -0,0 +1,92 @@
+import { useState } from 'react';
+import { Column } from '../../types';
+import { MIN_COLUMN_WIDTH, ColumnWidthTypes } from '../constants';
+import useDidUpdateEffect from './use-did-update-effect';
+import useMeasureText, { MeasureTextHook } from '../virtualized-table/hooks/use-measure-text';
+
+/**
+ * Calculates column widths in pixels, based on column settings and the table width.
+ * First, pixel values for the three independent types ('pixels', 'percentage', 'hug') are set.
+ * Then the remaining width is divided equally between the 'fill' columns, if there is any
+ * The widths are sorted in the order they will be displayed
+ */
+export const getColumnWidths = (
+  columns: Column[],
+  tableWidth: number,
+  { measureText, estimateWidth }: MeasureTextHook
+) => {
+  if (!columns?.length || tableWidth === 0) return [];
+
+  const columnWidths: number[] = [];
+  const fillColumnIndexes: number[] = [];
+  let sumFillWidths = tableWidth;
+
+  columns.forEach((col, idx) => {
+    if (col.columnWidth) {
+      const {
+        columnWidth: { type, pixels, percentage },
+        label,
+        qApprMaxGlyphCount,
+      } = col;
+      let newWidth = 0;
+
+      const addKnownWidth = () => {
+        columnWidths[idx] = Math.max(MIN_COLUMN_WIDTH, newWidth);
+        sumFillWidths -= columnWidths[idx];
+      };
+
+      switch (type) {
+        case ColumnWidthTypes.PIXELS:
+          if (!pixels) throw new Error(`pixel value is not defined`);
+          newWidth = pixels;
+          addKnownWidth();
+          break;
+        case ColumnWidthTypes.PERCENTAGE:
+          if (!percentage) throw new Error(`pixel value is not defined`);
+          newWidth = (percentage / 100) * tableWidth;
+          addKnownWidth();
+          break;
+        case ColumnWidthTypes.HUG:
+          newWidth = Math.max(measureText(label), estimateWidth(qApprMaxGlyphCount));
+          addKnownWidth();
+          break;
+        case ColumnWidthTypes.FILL:
+        case undefined:
```

Comment:
for backwards compability. We are not sure that columnWidth exists on the layout, and then it wont have a type defined

Reply:
We might actually set it optional in the API as well. If you are building a mashup and writing the hypercube yourself, it would be tediuos to set the column width for every column, when you want  it to be auto

Reply:
hmm actually. we check if `column.columnWidth` is defined, before the switch. So it would kind of make sense that you would get to the defualt case of throwing an error if you have a `columnWidth` object without a type defined. Will remve the undefined case here

Reply:
done

Reply:
if  `columnWidth` may not be on the layout, could we move that as the default behavior of this switch case then?

Reply:
So with my change it is like this: if columnWidth is not defined at all, we go to the else block, using the auto option which is fill for now. If columnWidth exists but not columnWidth.type, we throw an error. In this case it will be `undefined is not a valid column width type`